src/documentation/runners/dataflow.md

```diff
@@ -61,44 +70,83 @@ gcloud auth application-default login
-  <td><code>tempLocation</code></td>
-  <td>Optional. Path for temporary files. If set to a valid Google Cloud Storage URL that begins with <code>gs://</code>, <code>tempLocation</code> is used as the default value for <code>gcpTempLocation</code>.</td>
+  <td>
+    <span class="language-java"><code>tempLocation</code></span>
+    <span class="language-python"><code>temp_location</code></span>
+  </td>
+  <td>
+    <span class="language-java">Optional.</span>
+    <span class="language-python">Required.</span>
+    Path for temporary files. Must be a valid Google Cloud Storage URL that begins with <code>gs://</code>.
+    <span class="language-java">If set, <code>tempLocation</code> is used as the default value for <code>gcpTempLocation</code>.</span>
+  </td>
   <td>No default value.</td>
 </tr>
-<tr>
+
+<!-- Only show for Java -->
+<tr class="language-java">
   <td><code>gcpTempLocation</code></td>
   <td>Cloud Storage bucket path for temporary files. Must be a valid Cloud Storage URL that begins with <code>gs://</code>.</td>
   <td>If not set, defaults to the value of <code>tempLocation</code>, provided that <code>tempLocation</code> is a valid Cloud Storage URL. If <code>tempLocation</code> is not a valid Cloud Storage URL, you must set <code>gcpTempLocation</code>.</td>
 </tr>
+
 <tr>
-  <td><code>stagingLocation</code></td>
+  <td>
+    <span class="language-java"><code>stagingLocation</code></span>
+    <span class="language-python"><code>staging_location</code></span>
+  </td>
   <td>Optional. Cloud Storage bucket path for staging your binary and any temporary files. Must be a valid Cloud Storage URL that begins with <code>gs://</code>.</td>
-  <td>If not set, defaults to a staging directory within <code>gcpTempLocation</code>.</td>
+  <td>
+    <span class="language-java">If not set, defaults to a staging directory within <code>gcpTempLocation</code>.</span>
+    <span class="language-python">If not set, defaults to a staging directory within <code>temp_location</code>.</span>
+  </td>
+</tr>
+
+<!-- Only show for Python -->
+<tr class="language-python">
+  <td><code>save_main_session</code></td>
+  <td>Save the main session state so that pickled functions and classes defined in <code>__main__</code> (e.g. interactive session) can be unpickled. Some workflows do not need the session state if, for instance, all of their functions/classes are defined in proper modules (not <code>__main__</code>) and the modules are importable in the worker.</td>
+  <td><code>false</code></td>
+</tr>
+
+<!-- Only show for Python -->
+<tr class="language-python">
+  <td><code>sdk_location</code></td>
+  <td>Override the default GitHub location from where the Cloud Dataflow SDK is downloaded. This value can be an URL, a Cloud Storage path, or a local path to an SDK tarball. Workflow submissions will download or copy the SDK tarball from this location. If set to the string <code>default</code>, a standard SDK location is used. If empty, no SDK is copied.</td>
```

Comment:
This description needs to be fixed in pipeline options as well. Could you change this to:
```
Override the default location from where the Beam SDK is downloaded. This value can be an URL, a Cloud Storage path, or a local path to an SDK tarball. Workflow submissions will download or copy the SDK tarball from this location. If set to the string <code>default</code>, a standard SDK location is used. If empty, no SDK is copied.
```